pkg/handler/repository_parameters.go

```diff
@@ -12,15 +12,49 @@ import (
 )
 
 type RepositoryParameterHandler struct {
-	RepositoryDao dao.RepositoryConfigDao
+	RepositoryDao       dao.RepositoryConfigDao
+	ExternalResourceDao dao.ExternalResourceDao
 }
 
-func RegisterRepositoryParameterRoutes(engine *echo.Group, repoDao *dao.RepositoryConfigDao) {
-	rph := RepositoryParameterHandler{RepositoryDao: *repoDao}
+func RegisterRepositoryParameterRoutes(
+	engine *echo.Group,
+	repoDao *dao.RepositoryConfigDao,
+	externalDao *dao.ExternalResourceDao,
+) {
+	rph := RepositoryParameterHandler{RepositoryDao: *repoDao, ExternalResourceDao: *externalDao}
 	engine.GET("/repository_parameters/", rph.listParameters)
+	engine.GET("/repository_parameters/external_gpg_key/", rph.fetchGpgKey)
```

Comment:
i kinda wonder if this should be a POST instead of a GET.   Since its fetching data from external servers, a mean person could embed a bunch of links on some page, which forces the visitor's browser to request those URLS.  This could slow our site, or potentially do something more malicious.

Reply:
i realized this while reviewing your frontend pr

Reply:
ahh... how would we prevent that with a POST??

Reply:
So for example, here's a get request example, create an html file with:

```
<html>
<img src="https://console.stage.redhat.com/api/content-sources/v1/ping"
</html>
```

If you login to console.stage.redhat.com within your browser and then browse to this html page, you'll see the ping request gets executed and you get a pong response.

If we try a similar thing with a post:

```
<html>

<script>

let xhr = new XMLHttpRequest();
xhr.open("POST", "https://console.stage.redhat.com/api/content-sources/v1/repositories/");
xhr.setRequestHeader("Accept", "application/json");
xhr.setRequestHeader("Content-Type", "application/json");

xhr.onreadystatechange = function () {
  if (xhr.readyState === 4) {
    console.log(xhr.status);
    console.log(xhr.responseText);
  }};

let data = `{
  "name": "fooobar34",
  "url": "http://yum.theforeman.org/pulpcore/3.14/el8/x86_64/",
}`;

xhr.send(data);
</script>

</html>
```                                     
It will fail due to cross site protection i believe, you get a forbidden with:  
Referrer Policy: strict-origin-when-cross-origin

Maybe this is related to the configuration of console dot.  But since i think this particular request probably shouldn't be cache-able, it also makes sense to be something other than a GET.